dipy/denoise/enhancement_kernel.pyx

```diff
@@ -0,0 +1,419 @@
+                     Combining Contextual PDE flow with Constrained Spherical 
+                     Deconvolution. PLoS One.
+        """
+
+        # save parameters as class members
+        self.D33 = D33
+        self.D44 = D44
+        self.t = t
+
+        # define a sphere
+        if type(orientations) is Sphere:
+            # use the sphere defined by the user
+            sphere = orientations
+        elif type(orientations) is int:
+            # electrostatic repulsion based on number of orientations
+            n_pts = orientations
+            theta = np.pi * np.random.rand(n_pts)
+            phi = 2 * np.pi * np.random.rand(n_pts)
+            hsph_initial  = HemiSphere(theta=theta, phi=phi)
+            sphere, potential = disperse_charges(hsph_initial , 5000)
+        else:
+            # use default
+            sphere = get_sphere('repulsion100')
+        self.orientations = sphere.vertices
+        self.sphere = sphere
+
+        # save openmp settings
+        self.num_threads = num_threads
+        
+        # file location of the lut table for saving/loading
+        kernellutpath = "%s/kernel_d33@%4.2f_d44@%4.2f_t@%4.2f_numverts%d.npy" \
+                        % (gettempdir(), D33, D44, t, len(self.orientations))
+
+        # create a lookup table in testing mode
+        if test_mode:
+            self.create_lookup_table(True)
+            return
+
+        # if LUT exists, load
+        if not force_recompute and os.path.isfile(kernellutpath):
+            print "The kernel already exists. Loading from " + kernellutpath
+            self.lookuptable = np.load(kernellutpath)
+
+        # else, create
+        else:
+            print "The kernel doesn't exist yet. Computing..."
+            self.create_lookup_table()
+            np.save(kernellutpath, self.lookuptable)
+            
+    def get_lookup_table(self):
+        """ Return the computed look-up table.
+        """
+        return self.lookuptable
+
+    def get_orientations(self):
+        """ Return the orientations.
+        """
+        return self.orientations
+        
+    def get_sphere(self):
+        """ Get the sphere corresponding with the orientations
+        """
+        return self.sphere
+
+    def evaluate_kernel(self, x, y, r, v):
+        return self.k2(x, y, r, v)
+
+    # Cython functions
+
+    @cython.wraparound(False)
+    @cython.boundscheck(False)
+    @cython.nonecheck(False)
+    @cython.cdivision(True)
+    cdef void create_lookup_table(self, test_mode = False):
+        """ Compute the look-up table based on the parameters set
+            during class initialization
+        """
+        self.estimate_kernel_size()
+
+        cdef:
+            double [:, :] orientations = np.copy(self.orientations)
+            int OR1 = orientations.shape[0]
+            int OR2 = orientations.shape[0]
+            int N = self.kernelsize
+            int hn = (N-1)/2
```

Comment:
For safety you can change all int to cnp.npy_intp so that you don't get into any troubles in the different operating systems and architectures.